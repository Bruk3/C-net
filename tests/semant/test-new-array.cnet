struct person {
    string name;
};

int main(){
    struct person john;
    struct person jane;
    int[4] myInts = new int[4]{1,2,3,4};
    string[2] myStrings = new string[2]{"hello", "world"};
    struct person[2] myPeople = new struct person[2]{john, jane};
}