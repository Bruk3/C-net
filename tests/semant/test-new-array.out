struct person {
	string name;
};

int main()
{
struct person john = (struct person : );
struct person jane = (struct person : );
int[] myInts = (int[] : );
(int[] : myInts=(int[] : new int[(int : 4)] = {(int : 1), (int : 2), (int : 3), (int : 4)}));
string[] myStrings = (string[] : new string[(int : 2)] = {(string : "hello"), (string : "world")});
struct person[] myPeople = (struct person[] : new struct person[(int : 2)] = {(struct person : john), (struct person : jane)});
return (int : 0);
}

